core/src/main/java/org/apache/metamodel/query/parser/SelectItemParser.java

```diff
@@ -160,6 +164,14 @@ public SelectItem findSelectItem(String expression) throws MultipleSelectItemsPa
                 return new SelectItem(subQuerySelectItem, fromItem);
             }
         }
+        
+        //if the expression is alias of some select item defined return clone of that select item
+        List<SelectItem> allSelectItems = MetaModelHelper.getAllSelectItems(_query);
```

Comment:
Also reflecting a bit on this ... Do we need to have this getAllSelectItems(...) method?

What I mean is, could there be aliased select items anywhere else than in the SELECT clause? I cannot imagine it really ... For instance in WHERE clause I have never seen anything a la:

SELECT ... FROM ... WHERE name AS n = "Kasper"

If we assume that this never happens then you can get rid of the getAllSelectItems(...) method which is already a bit greedy (makes new lists etc.). If I am wrong about the assumption (I seriously am in doubt) then I think the solution is good!